File: data_structures/adjacency_list_graph/python/test_adjacency_list_graph.py
import pytest
from adjacency_list_graph import AdjacencyListGraph


def test_add_vertex():
    g = AdjacencyListGraph()
    g.add_vertex(0)
    g.add_vertex(1)
    g.add_vertex(2)
    assert g.vertex_count() == 3
    assert g.edge_count() == 0


def test_add_edge():
    g = AdjacencyListGraph()
    g.add_edge(0, 1)
    g.add_edge(0, 2)
    g.add_edge(1, 2)
    assert g.has_edge(0, 1)
    assert g.has_edge(0, 2)
    assert not g.has_edge(2, 0)
    assert g.edge_count() == 3
    assert g.vertex_count() == 3
    nb = g.get_neighbors(0)
    assert 1 in nb and 2 in nb


def test_auto_creates_vertices():
    g = AdjacencyListGraph()
    g.add_edge(5, 10)
    assert g.has_vertex(5)
    assert g.has_vertex(10)


def test_bad_vertex_raises():
    g = AdjacencyListGraph()
    with pytest.raises(KeyError):
        g.get_neighbors(99)

File: data_structures/adjacency_list_graph/matlab/test_adjacency_list_graph.m
% test_adjacency_list_graph.m  (頂点IDは0始まり)

g = adjacency_list_graph('new', 20);
g = adjacency_list_graph(g, 'add_vertex', 0);
g = adjacency_list_graph(g, 'add_vertex', 1);
g = adjacency_list_graph(g, 'add_vertex', 2);
assert(adjacency_list_graph(g, 'vertex_count') == 3);
assert(adjacency_list_graph(g, 'edge_count')   == 0);

g = adjacency_list_graph(g, 'add_edge', 0, 1);
g = adjacency_list_graph(g, 'add_edge', 0, 2);
g = adjacency_list_graph(g, 'add_edge', 1, 2);
assert(adjacency_list_graph(g, 'has_edge', 0, 1));
assert(adjacency_list_graph(g, 'has_edge', 0, 2));
assert(~adjacency_list_graph(g, 'has_edge', 2, 0));
assert(adjacency_list_graph(g, 'edge_count') == 3);

nb = adjacency_list_graph(g, 'get_neighbors', 0);
assert(any(nb == 1) && any(nb == 2));

try
    adjacency_list_graph(g, 'get_neighbors', 15);
    error('expected error not thrown');
catch e
    assert(strcmp(e.identifier, 'adjacency_list_graph:noVertex'));
end

disp('All tests passed.');
